### PR TITLE
Add more search options

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This adds more fields to the filters on search for individual
- organization_name (Employer Name)
- first_name
- nick_name
- last_name

Note
1) organization_name is populated for individuals with employers - it's handy to be able
to filter by as a quick way to narrow down. I wasn't too worried about declaring it only
for individuals as sort_name covers this for organizations
2) Last name these feels a bit unnecessary as it co-incides with sort_name
but it also feels odd not to add it
3) Ideally for these text fields they would be always or optionally 'like' searches
(respecting the site +wildcard setting). I took a a quick look but decided to leave it
out of scope at this stage.
4) I thought 'email' might have been useful too but given 3 & the fact that the main use I saw
was as a proxy for 1 (employer) and that email is the preferred search for email anyway (
which was the entity I was working on) I excluded it too in the end.

The use case for this is narrowing down in a large database. Take finding Johnathan Smith.
Being able to add 'Johnathan' as a filter helps get past all the Johns quicker.

Lastly - I have a sneaky feeling that users in our org will abuse the nick name filter heavily &
add nicknames to all contacts they frequently email....

Before
----------------------------------------
Screen Shot 2020-05-21 at 3 56 29 PM

After
----------------------------------------
Screen Shot 2020-05-21 at 3 55 24 PM

Technical Details
----------------------------------------

Comments
----------------------------------------

@colemanw - I haven't quite tackled the to convert yet - it's a bit scary so I just did some more cleanup steps in that direction for now - but I tested cc & bcc on staging today & performance is good

### CODE REVIEW

File: CRM/Contact/BAO/Contact.php [hunk 1]
```diff
@@ -3720,6 +3720,10 @@ public static function getEntityRefFilters() {
       ['key' => 'tag', 'value' => ts('Tag'), 'entity' => 'EntityTag'],
       ['key' => 'state_province', 'value' => ts('State/Province'), 'entity' => 'Address'],
       ['key' => 'country', 'value' => ts('Country'), 'entity' => 'Address'],
+      ['key' => 'first_name', 'value' => ts('First Name'), 'type' => 'text', 'condition' => ['contact_type' => 'Individual']],
+      ['key' => 'last_name', 'value' => ts('Last Name'), 'type' => 'text', 'condition' => ['contact_type' => 'Individual']],
+      ['key' => 'nick_name', 'value' => ts('Nick Name'), 'type' => 'text', 'condition' => ['contact_type' => 'Individual']],
+      ['key' => 'organization_name', 'value' => ts('Employer name'), 'type' => 'text', 'condition' => ['contact_type' => 'Individual']],
```
Comment (seamuslee001) [inline CRM/Contact/BAO/Contact.php:3726]: should the contact type be org here? or is it a different field we are searing in?

Comment (eileenmcnaughton) [reply CRM/Contact/BAO/Contact.php:3726]: @seamuslee001 check the PR comments - in this case it is called 'Employer name' & works for individuals. I figured not having it for orgs was OK as it basically IS the sort name that is searched anyway

**Comment by yashodha** [comment]
@eileenmcnaughton 
I think we could have a few more fields, at the very least the filters used for Quick Search.
search

**Comment by eileenmcnaughton** [comment]
@yashodha I left out the ones that aren't directly contact related for now because the image above is the entity reference field for emails - not for contacts. It's on the email send task. Email is the primary field it searches on so it's actually clumsier adding it. I'd like the address fields but the many to one join means they need more testing for any performance issues if they are added - since I guess a group by would be needed too. So I left them out of scope for now.

**Comment by eileenmcnaughton** [comment]
Having said all that - for the contact entity reference field it looks like street address, city & postal code might be easy adds - I'll check that!

**Comment by eileenmcnaughton** [comment]
@yashodha  OK - I added City, postal code, street address.

I have some misgivings about street address. The issue is that only equals is supported at the moment so for both street address & job title it might create more confusion as people can't find 'st' vs 'Street' (I've almost convinced myself to remove it again...)

Screen Shot 2020-05-29 at 9 44 41 AM

**Comment by colemanw** [comment]
I tend to agree that street address is not suitable for this format and (at least with the current impl) will cause more confusion than benefit.

**Comment by eileenmcnaughton** [comment]
@colemanw I took out street address againn

**Comment by yashodha** [comment]
@eileenmcnaughton great, thanks!

**Pull request merged**